src/fs-cache.js

```diff
@@ -125,35 +125,36 @@ module.exports = function(params, callback) {
   const identifier = params.identifier;
   let directory;
 
-  if (typeof params.directory === "string") {
-    directory = params.directory;
-  } else {
-    directory = findCacheDir({ name: "babel-loader" }) || os.tmpdir();
+  try {
+    if (typeof params.directory === "string") {
+      directory = params.directory;
+      mkdirp.sync(directory);
+    } else {
+      directory = findCacheDir({ name: "babel-loader", create: true });
```

Comment:
If `findCacheDir` doesn't find the package root directory, it doesn't throw, only returns `null`. Can you handle that case below? If directory is `null` we should fall back to ` os.tmpdir()`.